types/src/database_sql.rs

```diff
@@ -637,6 +637,25 @@ impl Database {
         Ok(bookmarks.into_iter().map(|x| x.id).collect())
     }
 
+    pub async fn get_document_bookmarked_on(
+        &self,
+        document_id: &Uuid,
+        user_id: &Uuid,
+    ) -> Result<Option<Date>> {
+        if let Some(bookmark) = query_file!(
+            "queries/get_document_bookmarked_on.sql",
+            document_id,
+            user_id
+        )
+        .fetch_optional(&self.client)
+        .await?
+        {
+            Ok(Some(date::Date(bookmark.bookmarked_on)))
+        } else {
+            Ok(None)
+        }
```

Comment:
I think you could clean this up with `query_file_scalar!` (this tells sqlx you are only fetching one column, and so it doesn't make a row struct for you) or by "mapping" over your option instead of doing `if let`. Also, I don't think this is worth holding you up over, this is just so you know about this going forward.

Here is the reference for query scalar (query file scalar just takes a file path instead) https://docs.rs/sqlx/latest/sqlx/macro.query_scalar.html 

 I think `query_file_scalar!` would be cleaner in this case but I think it's worth showing what the solution with map would look like if you're ever in this situation again. Here it is with map:

```suggestion
        Ok(
        query_file!(
            "queries/get_document_bookmarked_on.sql",
            document_id,
            user_id
        )
        .fetch_optional(&self.client)
        .await?
        .map(|bookmark| date::Date(bookmark.bookmarked_on))
        )
```

Here are the docs for "map" on the Option type. https://doc.rust-lang.org/std/option/enum.Option.html#method.map